src/operator/controllers/external_traffic/service_uploader.go

```diff
@@ -53,22 +83,92 @@ func (s *ServiceUploaderImpl) UploadNamespaceServices(ctx context.Context, names
 		return errors.Wrap(err)
 	}
 
+	s.namespaceToReportedServicesCache.Add(namespace, hashSum)
+
 	return nil
 }
 
-func (s *ServiceUploaderImpl) getExternalService(ctx context.Context, svc *corev1.Service) (graphqlclient.ExternallyAccessibleServiceInput, bool, error) {
+func reportedServicesCacheValuePart(namespace, serverName string) []byte {
+	return []byte(fmt.Sprintf("%s%s", serverName, namespace))
+}
+
```

Comment:
I think it's better to print here the whole struct, to make sure any change in any field, also on future updates, will cause upload.
```suggestion

func reportedServicesCacheValuePart(service graphqlclient.ExternallyAccessibleServiceInput) []byte {
	return []byte(fmt.Sprintf("\"%+v", service))
}

```